SDK v2 migration guide.md

```diff
@@ -0,0 +1,399 @@
+are outlines of the notable breaking changes as well as a mapping from v1 APIs to v2 APIs to aid you in migrating.
+
+## Table Of Contents
+
+ - [Why the v1 SDK is being replaced](#Why-the-v1-sdk-is-being-replaced)
+ - [What will happen to the v1 SDK](#What-will-happen-to-the-v1-sdk)
+ - [Migration Guide](#migration-guide)
+   - [IoT Hub Device Client](#iot-hub-device-client)
+   - [IoT Hub Service Client](#iot-hub-service-client)
+   - [DPS Device Client](#dps-device-client)
+   - [DPS Service Client](#dps-service-client)
+   - [IoT Hub Device Client](#iot-hub-device-client)
+   - [Security Provider Clients](#security-provider-clients)
+   - [Deps](#deps)
+ - [Frequently asked questions](#frequently-asked-questions)
+
+## Why the v1 SDK is being replaced
+
+There are a number of reasons why the Azure IoT SDK team chose to do a major version revision. Here are a few of the more important reasons:
+  - [Upcoming certificate changes](./upcoming_certificate_changes_readme.md) dictated that the SDK needed to stop pinning on a specific IoT Hub public certificate and start reading certificates from the device certificate store.
+  - Several 3rd party dependencies (Bouncycastle, JNR Unixsocket, Azure Storage SDK) were becoming harder to carry due to security concerns and they could only be removed by removing or alterring existing APIs.
+  - Many existing client classes (RegistryManager, DeviceTwin, DeviceMethod, ServiceClient, etc.) were confusingly named and contained methods that weren't always consistent with the client's assumed responsibilities.
+  - Many existing client's had a mix of standard constructors (```new DeviceClient(...)```) and static builder constructors (```DeviceClient.createFromSecurityProvider(...)```) that caused some confusion among users.
+  - ```DeviceClient``` and ```ModuleClient``` had unneccessarily different method names for the same operations (```deviceClient.startDeviceTwin(...)``` vs ```moduleClient.startTwin(...)```) that could be easily unified for consistency.
+  - ```DeviceClient``` and ```ModuleClient``` had many asynchronous methods whose naming did not reflect that they were asynchronous. This led to some users calling these methods as though they were synchronous.
+
+## What will happen to the v1 SDK
+
+We have released [one final LTS version](https://github.com/Azure/azure-iot-sdk-java/releases/tag/2022-03-04) of the v1 SDK that
+we will support like any other LTS release (security bug fixes, some non-security bug fixes as needed), but users are still encouraged
+to migrate to v2 when they have the chance.
+
+## Migration Guide
+
+### IoT Hub Device Client
+
+#### DeviceClient
+
+| V1 class#method  | Equivalent V2 class#method |
+|:---|:---|
+| DeviceClient#createFromSecurityProvider(String, String, SecurityProvider, IotHubClientProtocol); | nwe DeviceClient(String, String, SecurityProvider, IotHubClientProtocol);  |
```

Comment:
Question - I forget, are we including defaulting to mqtt as a part of the v2 changes?

Reply:
We don't have a default protocol for v2, no. Users always have to pick one explicitly.